### ISSUE TITLE
Development Environment Setup

### ISSUE THREAD

**Issue by pyscripter99** [mfts/papermark#449]
In it's current state, Papermark does not have clear instructions for setting up a development environment. Over the past few weeks I have been trying over and over to get Papermark in a development ready state, with minimal success. I hope that someone in the community can assist in creating an appropriate Docker setup.

**Comment by mfts** [comment]
Fair point. We will be publishing a proper self-hosting guide later this month. 

Thanks for bringing it up :)

**Comment by bndrgroup** [comment]
+1 for a docker image - would love to host this within k8s.

**Comment by jo-chemla** [comment]
👀 

Watching, also interested to self-host via a container image build that can be used within a docker-compose stack, on ghcr or docker-hub registries!

**Comment by Flyingfufu** [comment]
+1 for a docker image

**Comment by bbenjamin-sys** [comment]
+1 for a docker image

**Comment by satoshinotdead** [comment]
Docker should bring more overall attention to this unique project!

**Comment by fan-wen** [comment]
Have been following this project for some time - a docker image will be deeply appreciated!!!

**Comment by xdemo1000** [comment]
Any updates on this?

**Comment by DipakKhade** [comment]
can any one tell me the requirements  for docker integration , so i can move forward with that

**Comment by harikrish-s** [comment]
Any updates on this? Requesting the contributors to update the readme with the latest setup instructions or an official docker setup

**Comment by rjherrera** [comment]
I agree! But even without docker, it would be great if you specified how you guys develop locally, cause for me it isn't clear from reading the readme file. 

The "copy the env file" step hides a lot of steps behind, and frankly, steps that don't look like they are specifically for development.

Awesome project though, I'd love to run this and contribute :)

**Comment by pyscripter99** [comment]
@mfts If you aren't already working on this, I'll take another crack at creating a development environment when I get the chance.

**Comment by bndrgroup** [comment]
I have managed to get this to work - with dockerfiles, env and kubernetes yaml's. 

Will make a PR when i have some time this week, as you have to do some undocumented things like create a trigger v2 project (which is no longer possible via the GUI)

**Comment by harikrish-s** [comment]
> I have managed to get this to work - with dockerfiles, env and kubernetes yaml's.
> 
> Will make a PR when i have some time this week, as you have to do some undocumented things like create a trigger v2 project (which is no longer possible via the GUI)

Hey, did you get time to work on this, if so could you please make the PR so that I can set it up locally and work on other issues

**Comment by satoshinotdead** [comment]
That will greatly increase the use of the app. I'm excited to try this out.

**Comment by JonasPetry** [comment]
Any update on this?

**Comment by zshzebra** [comment]
> Any update on this?

Apparently work had been done, but I'm not sure if a PR was ever made.

> I have managed to get this to work - with dockerfiles, env and kubernetes yaml's. 
> 
> Will make a PR when i have some time this week, as you have to do some undocumented things like create a trigger v2 project (which is no longer possible via the GUI)

**Comment by satonotdead** [comment]
That should bring a lot of new users to the open-source app :)

**Comment by klarlichter** [comment]
Hi , did some have some working docker file that it is more easy to install ? And is that environment exactly with the same function as on their page ? 
Is there somewhere a possibility to setup smtp for send that invation mail ?

**Comment by askprateek** [comment]
Waiting for this.

**Comment by RowanAldean** [comment]
Reminder that this is important for organisations!

**Comment by satonotdead** [comment]
> Reminder that this is important for organisations!

And everyone, every single human on this planet needs to respect their own freedom (privacy).

**Comment by satonotdead** [comment]
May we know from an experienced developer what's wrong with #533?

**Comment by s4wyer161** [comment]
+1 for a docker image

**Comment by icardo** [comment]
After a quick search on this, I viewed a Reddit thread on the same and was referred here. Over a year later and no progress on this.
Seeing that there's a paid hosting option, it seems there's no real interest by the org. in actually finalizing an easy, streamlined method to self host. I could be wrong.
I see a PR, but nothing materializing out of this.

**Comment by satonotdead** [comment]
I ended up letting go Papermark on my stack.

**Comment by kuna1bajaj** [comment]
Is there any progress on this? It seems like folks are willing to work on the docker image but the repo owner doesn’t want it

**Comment by mdr-github** [comment]
I don't expect this to happen

**Comment by s4wyer161** [comment]
Terrible founder. Calling it open source just for the label. He can’t even make clear what’s truly free and what’s not. It’s obvious he has zero real interest in the open-source community.

**Comment by kuna1bajaj** [comment]
> That should bring a lot of new users to the open-source app :)

I dont think the founder wants more users on open source, he seems to want free devs contributing to the project.

**Comment by amitshuklabag** [comment]
It has been over a year since the last update. Could you please share any recent developments on this topic?

**Comment by MrDaggubati** [comment]
I bet there is never going to be a Dockerfile which one could use to self-host.
Also, it's not fair to expect when even big organisations are hellbent on feasting on free lunch on open software without any commitment or support.

**Comment by satonotdead** [comment]
There are a lot of expectations about this issue. Just with this app the best.

**Comment by riazahmed2246** [comment]
Docker installation needed. Manual installation showing errors maintaining proper instructions.

**Comment by aismaniotto** [comment]
+1
